src/main/java/health/ere/ps/retry/Retrier.java

```diff
@@ -0,0 +1,45 @@
+package health.ere.ps.retry;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+
+public class Retrier {
+
+    private Retrier() {
+    }
+
+    public static <T> T callAndRetry(
+        List<Integer> retrySeconds,
+        int retryPeriodMs,
+        RetryAction<T> action,
+        Predicate<T> predicate
+    ) {
+        T result = action.execute();
```

Comment:
You should catch an Throwable here and log it. Otherwise it might be swalloed.

Reply:
Fixed